shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/core/data/PlantMapper.kt

```diff
@@ -17,4 +18,22 @@ fun String.toPlants(): List<Plant> {
     } catch (e: Exception) {
         return emptyList()
     }
-}
\ No newline at end of file
+}
+
+fun Long.toPlant(plants: List<Plant>): Plant {
```

Comment:
I think you can **get rid of it**, because it pointless. You can't use this obj, you will need an extra round to find the "real" obj which belongs to it's id.

I show you an example how can you use it in repository if you don't mind ->
Presume you have a flow with favourite ids like this ->
val favouriteIds: Flow<List<Long>>

And you have all the plants ->
val plants: List<Plant>

You can map them like this ->

If you want a flow with favorite plants ->

favouriteIds.map { ids ->
        ids.mapNotNull { id ->
            plants.firstOrNull { it.id.toLong() == id }
        }
    }

If you want set isFavourite (do it in suspend fun) ->
plants.map { 
      it.isFavorite = favouriteIds.first().contains(it.id.toLong())
      it
}

Of course you can use your solution as well :) these were some example

Reply:
Refactored. Thanks for the samples :)